src/app.rb

```diff
@@ -0,0 +1,93 @@
+require './person'
+require './trimmer_decorator'
+require './capitalize_decorator'
+require './book'
+require './rental'
+require './student'
+require './teacher'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+  end
+
+  def list_books
+    @books.each do |book|
+      puts "#{book.title} by #{book.author} on #{book.rentals.count} rentals"
+    end
+  end
+
+  def list_people
+    @people.each do |person|
+      puts "#{person.name} - #{person.class} - #{person.id}"
+    end
+  end
+  # create person.......................
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? '
+    person_type = gets.chomp.to_i
+    print 'Name:'
+    name = gets.chomp
+    print 'Age:'
+    age = gets.chomp.to_i
+    if person_type == 1
+      print 'Whats the parent permission? (y/n) '
+      parent_permission = gets.chomp == 'y'
+      @people << Student.new(name, age, parent_permission)
+      puts 'Student created successfully!'
+    elsif person_type == 2
+      print 'Specialization:'
+      specialization = gets.chomp
+      @people << Teacher.new(name, age, specialization)
+      puts 'Teacher created successfully!'
+    else
+      puts 'Person not created !'
+    end
+  end
+  # create books.......................
+
+  def create_book
+    print 'Title:'
+    title = gets.chomp
+    print 'Author:'
+    author = gets.chomp
+    @books << Book.new(title, author)
+    puts 'Book created successfully!'
+  end
+
+  # creata rental.......................
+
+  def create_rental
+    puts 'Choose a book from the list by number'
+    @books.each_with_index do |book, index|
+      puts "#{index} - #{book.title} by #{book.author}"
+    end
+    book_index = gets.chomp.to_i
+    book = @books[book_index]
+    puts 'Choose a person from the list by number:'
+    @people.each_with_index do |person, index|
+      puts "#{index} - #{person.name} - #{person.age} - #{person.class}"
+    end
+    person_index = gets.chomp.to_i
+    person = @people[person_index]
+    puts 'Date: '
+    date = gets.chomp
+    Rental.new(date, book, person)
+    puts 'Rental created successfully!'
+  end
+  # create list_rentals.......................
+
+  def list_rentals
+    puts 'Choose a person from the list by number'
+    @people.each_with_index do |person, index|
+      puts "#{index} - #{person.name}"
+    end
+    person_index = gets.chomp.to_i
+    person = @people[person_index]
+    person.rentals.each do |rental|
+      puts "Book: #{rental.book.title} - Date: #{rental.date}"
+    end
+  end
+end
```

Comment:
- [ ]  When the option 6 is selected to list all rentals, you are required to provide the id of the person as shown in the image below, however you are requiring the number of the person, please kindly make the necessary changes and consult the [video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) for reference. :+1: :sparkles:
- project requirement
![num](https://github.com/fatima-najafi/oop-school-library/assets/89055502/319fb950-4e49-4c7a-a70d-b65e3271eb5c)
- your design
![num](https://github.com/fatima-najafi/oop-school-library/assets/89055502/11efa900-905e-42d9-b832-d11af3c6ff25)

- [ ] After providing the ID this is how the rental information should be displayed as shown in the image below
![rent](https://github.com/fatima-najafi/oop-school-library/assets/89055502/92319cc4-5a4e-445f-841b-9f0b73825fb2)